conftest.py

```diff
@@ -21,10 +21,11 @@ def _py_files(folder):
     *_py_files("tests/CrawlerRunner"),
 ]
 
-for line in open('tests/ignores.txt'):
-    file_path = line.strip()
-    if file_path and file_path[0] != '#':
-        collect_ignore.append(file_path)
+with open('tests/ignores.txt') as reader:
+    for line in reader:
+        file_path = line.strip()
+        if file_path and file_path[0] != '#':
+            collect_ignore.append(file_path)
```

Comment:
Is this change necessary? I like the style of your version better, but if it’s just an style change, given the extend of this pull request, maybe it would be best to minimize unrelated style changes.

Reply:
Yes, without `with` in raises `ResourceNotReleased` warning on pypy interpreter. It breaks some tests.

Reply:
I created a separate [PR with test fixes](https://github.com/scrapy/scrapy/pull/5160)